alert-banner-dex-commented cookies

vpandura opened the pull request:
### Summary
Summarize the contents of the code changes in your pull request. Tag any open issues you believe to be resolved by this pull request.

### Additional Details
If you have anything else that you think may be relevant to this issue, list it here. Additional information can help us better understand your changes and speed up the review process.

Please add your name to the [CONTRIBUTORS.md] file. Adding your name to the [CONTRIBUTORS.md] file signifies agreement to all rights and reservations provided by the [License].

Thanks for contributing to Terra. 
@cerner/terra

[CONTRIBUTORS.md]: ../blob/master/CONTRIBUTORS.md
[License]: ../blob/master/LICENSE

Commit:
alert-banner-dex-commented cookies

chris-bert commented on packages/terra-consumer-layout/src/Layout.jsx:
```diff
@@ -1,5 +1,7 @@
 import React from 'react';
 import PropTypes from 'prop-types';
+import Alert from 'terra-alert';
+//import Cookies from 'js-cookie';
```
Remove commented out line

chris-bert commented on packages/terra-consumer-layout/src/Layout.jsx:
```diff
@@ -41,9 +51,19 @@ class Layout extends React.Component {
+  //       { alerts: prevState.alerts.filter(alert => (alert.id !== messageID)) },
+  //     ));
+  //   };
+  // }
```
Remove commented out lines above

chris-bert commented on packages/terra-consumer-layout/tests/nightwatch/DexLayout.jsx:
```diff
@@ -129,6 +129,16 @@ const data = {
+      messageText: 'This is a test' },
+    {
+      messageID: 'null',
+      messageText: 'This is a test -null' },
```
Perhaps include a hyperlink in the second alert.

Commit:
fixed issues

chris-bert commented on packages/terra-consumer-layout/package.json:
```diff
@@ -38,7 +38,9 @@
-    "terra-responsive-element": "^1.7.0"
+    "terra-responsive-element": "^1.7.0",
+    "terra-alert": "^1.9.1",
+    "js-cookie": "2.1.3"
```
Remove js-cookie since it is not used here.

chris-bert commented on packages/terra-consumer-layout/src/Layout.jsx:
```diff
@@ -18,6 +19,16 @@ const propTypes = {
+    alertList: PropTypes.arrayOf(
+      PropTypes.shape({
+        messageID: PropTypes.string,
+        messageText: PropTypes.oneOfType([
```
You should be able to use PropTypes.node here like the terra-alert does instead of specifying string and element types.

chris-bert replied:
nit: Maybe change these props to alertId and alertMessage since more than just text is supported.

Commit:
Addressed comments

Commit:
dismiss button event

Commit:
fixed lint errors

chris-bert commented on packages/terra-consumer-layout/src/Layout.jsx:
```diff
@@ -41,9 +45,8 @@ class Layout extends React.Component {
-
   render() {
-    const { nav, helpItems, intl, ...customProps } = this.props;
+    const { nav, helpItems, intl, siteAlert, ...customProps } = this.props;
```
nit: move siteAlert prop before intl prop

Commit:
changed log & lint errors

Commit:
tests

Commit:
translations

Commit:
fixed build errors

Commit:
lint errors

aberge00 commented on packages/terra-consumer-layout/src/Layout.jsx:
```diff
@@ -41,9 +45,8 @@ class Layout extends React.Component {
-
   render() {
-    const { nav, helpItems, intl, ...customProps } = this.props;
+    const { nav, helpItems, intl, siteAlert, ...customProps } = this.props;
```
I just confirmed with our SD that we want the AlertBanner to be positioned across the entire top of the window, extending over the Nav.

aberge00 replied:
Would like to have a visible separation between individual sitAlert elements.

aberge00 replied:
Can we attach a screen capture for desktop and mobile views.

aberge00 replied:
to better represent how we expect to use the component from within DEx, can you add  
    type={Alert.Opts.Types.INFO}   to the alert props.

![image](https://user-images.githubusercontent.com/13628009/32116040-117cf16a-bb17-11e7-8358-34d9208a8ac7.png)

aberge00 commented on packages/terra-consumer-layout/CHANGELOG.md:
```diff
@@ -1,5 +1,14 @@
+### Changed
+- Updating layout component to include terra-alert component
+- Updating DexLayout to include site alert example
+- terra-alert component used for site alert
```
Can you clarify what changes are to prod, and what's for tests.  ( Prod, only layout changed).

Also, can you call out the changes to Nav that was a separate bug fix.

Commit:
Changes in log file & siteAlert position

Commit:
changes in example page

alex-bezek committed:
upgrade and fix pop up dependency (#92)

* upgrade and fix pop up dependency

* add chagnelog entry

* update to use the newly exposed opts from terra popup instead of cheating and using the internal file

Commit:
Merge branch 'master' into DEX-alertbanner-vaishali

Commit:
Merge branch 'master' into DEX-alertbanner-vaishali

Commit:
pulled changes

chris-bert commented on packages/terra-consumer-layout/tests/nightwatch/DexLayout.jsx:
```diff
@@ -137,7 +137,7 @@ export default class DexLayout extends React.Component {
     super();
     this.state = {
-      alerts: ['001', '002'],
+      alerts: ['001'],
```
I would keep the two alerts here to show that multiple alerts are supported.

Commit:
added alert banner

Commit:
changes

Commit:
nit

kafkahw committed:
Layout mobile logo (#93)

* Add capability for center justified logo showing up in mobile viewports.

Commit:
update changelogs

Commit:
Publish

 - terra-consumer-layout@0.2.0
 - terra-consumer-nav@0.2.6
 - terra-consumer-site@0.4.10

Commit:
Merge branch 'master' into DEX-alertbanner-vaishali

mhemesath committed:
Fix mobile nav border styles (#89)

* Fix mobile nav border styles

* Update .travis.yml

* Updated changelog

kafkahw committed:
Fixing the center logo on mobile (#97)

* Fixing the center logo on mobile

Commit:
update changelog

Commit:
Publish

 - terra-consumer-layout@0.2.1
 - terra-consumer-nav@0.2.7
 - terra-consumer-site@0.4.11

kafkahw committed:
Fixing burger bar without icon (#98)

* Fixing burger bar without icon

* fixes test issues

* fixing sass linter error

Commit:
Publish

 - terra-consumer-layout@0.2.2
 - terra-consumer-site@0.4.12

Commit:
text spacing

Commit:
Merge branch 'DEX-alertbanner-vaishali' of https://github.com/vpandura/terra-consumer into DEX-alertbanner-vaishali

Commit:
Merge branch 'master' into DEX-alertbanner-vaishali

Commit:
addressed comments & lint

Commit:
lint
